fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/AccessControllerManager.java

```diff
@@ -0,0 +1,161 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.mysql.privilege;
+
+import org.apache.doris.analysis.TableName;
+import org.apache.doris.analysis.UserIdentity;
+import org.apache.doris.catalog.AuthorizationInfo;
+import org.apache.doris.datasource.InternalCatalog;
+import org.apache.doris.mysql.privilege.Auth.PrivLevel;
+import org.apache.doris.qe.ConnectContext;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+
+/**
+ * AccessControllerManager is the entry point of privilege authentication.
+ * There are 2 kinds of access controller:
+ * SystemAccessController: for global level priv, resource priv and other Doris internal priv checking
+ * CatalogAccessController: for specified catalog's priv checking, can be customized.
+ * And using InternalCatalogAccessController as default.
+ */
+public class AccessControllerManager {
+    private SystemAccessController sysAccessController;
+    private CatalogAccessController internalAccessController;
+    private Map<String, CatalogAccessController> ctlToCtlAccessController = Maps.newConcurrentMap();
+
+    public AccessControllerManager(Auth auth) {
+        sysAccessController = new SystemAccessController(auth);
+        internalAccessController = new InternalCatalogAccessController(auth);
+        ctlToCtlAccessController.put(InternalCatalog.INTERNAL_CATALOG_NAME, internalAccessController);
+    }
+
+    private CatalogAccessController getAccessControllerOrDefault(String ctl) {
+        return ctlToCtlAccessController.getOrDefault(ctl, internalAccessController);
+    }
+
+    public void addCatalogAccessControl(String ctl, CatalogAccessController controller) {
+        ctlToCtlAccessController.put(ctl, controller);
+    }
+
+    public Auth getAuth() {
+        return sysAccessController.getAuth();
+    }
+
+    // ==== Global ====
+    public boolean checkGlobalPriv(ConnectContext ctx, PrivPredicate wanted) {
+        return checkGlobalPriv(ctx.getCurrentUserIdentity(), wanted);
+    }
+
+    public boolean checkGlobalPriv(UserIdentity currentUser, PrivPredicate wanted) {
+        return sysAccessController.checkGlobalPriv(currentUser, wanted);
+    }
+
+    // ==== Catalog ====
+    public boolean checkCtlPriv(ConnectContext ctx, String ctl, PrivPredicate wanted) {
+        return checkCtlPriv(ctx.getCurrentUserIdentity(), ctl, wanted);
+    }
+
+    public boolean checkCtlPriv(UserIdentity currentUser, String ctl, PrivPredicate wanted) {
+        boolean hasGlobal = sysAccessController.checkGlobalPriv(currentUser, wanted);
+        boolean hasCtl = getAccessControllerOrDefault(ctl).checkCtlPriv(hasGlobal, currentUser, ctl, wanted);
+        return hasGlobal || hasCtl;
```

Comment:
logic is repeated with 'CatalogAccessController.checkCtlPriv(boolean hasGlobal, UserIdentity currentUser, String ctl, PrivPredicate wanted)',here,can we use 'CatalogAccessController.checkCtlPriv(UserIdentity currentUser, String ctl, PrivPredicate wanted)'? and delete 'CatalogAccessController.checkCtlPriv(boolean hasGlobal, UserIdentity currentUser, String ctl, PrivPredicate wanted)'?

Reply:
No exactly,
`CatalogAccessController` can be customized, and some implementation need to audit all request.
So we need to make sure that `CatalogAccessController` is called for each request.
And `sysAccessController` is used to check global level privilege(which is not included in `CatalogAccessController`)